assets/js/extensions/shipping-methods/pickup-location/edit-location/form.tsx

```diff
@@ -0,0 +1,133 @@
+/**
+ * External dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { SelectControl, TextControl } from '@wordpress/components';
+import { getSetting } from '@woocommerce/settings';
+
+/**
+ * Internal dependencies
+ */
+import type { PickupLocation } from '../types';
+import StateControl from './state-control';
+
+const Form = ( {
+	formRef,
+	values,
+	setValues,
+}: {
+	formRef: React.RefObject< HTMLFormElement >;
+	values: PickupLocation;
+	setValues: React.Dispatch< React.SetStateAction< PickupLocation > >;
+} ) => {
+	const countries = getSetting< Record< string, string > >( 'countries', [] );
+	const states = getSetting< Record< string, Record< string, string > > >(
+		'states',
+		[]
+	);
+	const setLocationField =
+		( field: keyof PickupLocation ) => ( newValue: string | boolean ) => {
+			setValues( ( prevValue: PickupLocation ) => ( {
+				...prevValue,
+				[ field ]: newValue,
+			} ) );
+		};
+
+	const setLocationAddressField =
+		( field: keyof PickupLocation[ 'address' ] ) =>
+		( newValue: string | boolean ) => {
+			setValues( ( prevValue ) => ( {
+				...prevValue,
+				address: {
+					...prevValue.address,
+					[ field ]: newValue,
+				},
+			} ) );
+		};
+
+	return (
+		<form ref={ formRef }>
+			<TextControl
+				label={ __( 'Location Name', 'woo-gutenberg-products-block' ) }
+				name={ 'location_name' }
+				value={ values.name }
+				onChange={ setLocationField( 'name' ) }
+				autoComplete="off"
+			/>
+			<TextControl
+				label={ __( 'Address', 'woo-gutenberg-products-block' ) }
+				name={ 'location_address' }
+				placeholder={ __( 'Address', 'woo-gutenberg-products-block' ) }
+				value={ values.address.address_1 }
+				onChange={ setLocationAddressField( 'address_1' ) }
+				autoComplete="off"
```

Comment:
any reason why auto-complete is off? even browser auto-complete?

Reply:
Yeah, this is browser auto-complete. It would not be ideal for your personal address to autocomplete your store address/pickup location in this context.